official/recommendation/ranking/configs/config.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ranking Model configuration definition."""
+from typing import Optional, List
+import dataclasses
+
+from official.core import exp_factory
+from official.modeling import hyperparams
+from official.modeling.hyperparams import config_definitions as cfg
+
+
+@dataclasses.dataclass
+class LearningRateConfig(hyperparams.Config):
+  """Learning rate scheduler config."""
+  learning_rate: float = 1.25
+  warmup_steps: int = 8000
+  decay_steps: int = 30000
+  decay_start_steps: int = 70000
+  decay_exp: float = 2
+
+
+@dataclasses.dataclass
+class OptimizationConfig(hyperparams.Config):
+  """Embedding Optimizer config."""
+  lr_config: LearningRateConfig = LearningRateConfig()
+  embedding_optimizer: str = 'SGD'
+
+
+@dataclasses.dataclass
+class DataConfig(hyperparams.Config):
+  """Dataset config for training and evaluation."""
+  input_path: str = ''
+  global_batch_size: int = 0
+  is_training: bool = True
+  dtype: str = 'float32'
+  shuffle_buffer_size: int = 10000
+  cycle_length: int = 10
+  sharding: bool = True
+  num_shards_per_host: int = 8
```

Comment:
Why do we need this?

Reply:
it's for[ “outer parallelism”](https://www.tensorflow.org/guide/data_performance_analysis#transformation_datasets) by running multiple copies of the input pipeline over sharded inputs and combining the results.